server/sonar-server/src/main/java/org/sonar/server/db/migrations/Select.java

```diff
@@ -105,6 +106,25 @@ public Date getDate(int columnIndex) throws SQLException {
     public byte[] getBytes(int columnIndex) throws SQLException {
       return rs.getBytes(columnIndex);
     }
+
+    @Override
+    public String toString() {
+      try {
+        ResultSetMetaData rsMetaData = rs.getMetaData();
+        StringBuilder sb = new StringBuilder();
+        for (int i = 1; i < rsMetaData.getColumnCount(); i++) {
+          if (i > 1) {
+            sb.append(",");
+          }
+          sb.append(rsMetaData.getColumnLabel(i).toLowerCase());
+          sb.append("=");
+          sb.append(rs.getObject(i));
+        }
+        return sb.toString();
+      } catch (Exception e) {
+        return "Unavailable: " + e.getMessage();
```

Comment:
Because I'm using row.toString() in MassUpdate to provide more details during an error. So this exception would hide the original one. As a general rule I think a toString method should never throw an exception. It's a pain during debugging in IDEs.